Feature/add models

perruche opened the pull request:
## Add 
- Add default Model for `Post` and `Page`
- Add `Timber\PostClassMap` filter to set the class map for Timber instantiation of Posts (closes #97)

## Update 
- Update `TaxonomiesManager` with reference to use `TaxonomyBuilder` from studiometa/wp-toolkit (closes #98)
- Update `CustomPostTypeManager` with reference to use `PostTypeBuilder` from studiometa/wp-toolkit

## Remove
- Remove unused `StudiometaPost` model

perruche committed:
Add Timber\PostClassMap filter to set the class map for Timber instantiation of Posts

perruche committed:
Add default Models for Post and Page

perruche committed:
Update CustomPostTypeManager with reference to PostTypeBuilder from studiometa/wp-toolkit

perruche committed:
Update TaxonomiesManager with reference to TaxonomyBuilder from studiometa/wp-toolkit

perruche committed:
Fix PHPStan error

perruche committed:
Fix PHPCS errors

notjb commented:
👏

titouanmathis commented on template/web/wp-content/themes/<%= slug %>/app/Managers/CustomPostTypesManager.php:
```diff
@@ -16,15 +17,52 @@ class CustomPostTypesManager implements ManagerInterface {
 	 * @return void
 	 */
-	public function register_custom_post_types() {
+	public function register_custom_post_types() {}
```
**question:** Should we rename this method to `register_custom_post_type()` to indicate that CPT should be registered in their own method?

perruche replied:
I'm not sure to understand, do you want to have one method for each post type ? like `register_post_type_product()`, `register_post_type_testimonies()` ?

titouanmathis replied:
Yes, that's the idea. I find the file easier to read and to manage to have one method per CPT.

perruche replied:
Should we have both ? To avoid adding multiple `add_action()`. 

Something like this :

```
class CustomPostTypesManager implements ManagerInterface {
	public function run() {
		add_action( 'init', array( $this, 'register_custom_post_types' ), 1 );
        }
       
      	public function register_custom_post_types() {
                 $this->register_post_type_product();
                 // Register other CPT bellow
        }

       public function register_post_type_product() {
                 // Register product CPT
       }
}
```

titouanmathis replied:
What is the impact on performance of multiple `add_action()` vs one with multiple method calls? 

I find the pattern with multiple `add_action()` less complex as it has one less level:

![mermaid-diagram-20211220112523](https://user-images.githubusercontent.com/250145/146752534-146738cd-f209-41bd-b97f-4f79a6abdc8e.png)

I think we should use the pattern you suggest only if there is a meaningful impact on performance with the usage of multiple `add_action`.

titouanmathis commented on template/web/wp-content/themes/<%= slug %>/app/Managers/TaxonomiesManager.php:
```diff
@@ -21,9 +22,11 @@ public function run() {
 	 */
-	public function register_taxonomies() {
-	}
+	public function register_taxonomies() {}
```
**question:** Same as for the CPT manager, should we rename this method to `register_taxonomies()`?

perruche committed:
Rename sample methods for register custom post type and taxonomy

- Add example usage from studiometa/wp-toolkit
- Add explicit comment to use one method per cpt/taxonomy register

perruche commented:
Latest comments from @titouanmathis are resolved, this PR is to review 😄

titouanmathis reviewed (approved):
LGTM! 👍
